File: app/src/main/java/com/storehop/app/data/db/relations/StorePickerItemRow.kt
package com.storehop.app.data.db.relations

import androidx.room.ColumnInfo

/**
 * One row per (item, store) pair that's relevant to the Store Picker badges
 * right now -- either currently needed at this store, OR purchased within
 * the active shopping session (so the picker can render an "✓ All set"
 * affirmation instead of a misleading "Nothing needed" once the user has
 * picked up everything that was on the list this trip).
 *
 * Multiple rows for the same item if it's tagged to multiple stores. The
 * picker repo groups by storeId and produces a
 * [com.storehop.app.data.repository.StorePickerRow] per live store.
 */
data class StorePickerItemRow(
    @ColumnInfo(name = "storeId") val storeId: String,
    @ColumnInfo(name = "itemId") val itemId: String,
    @ColumnInfo(name = "itemName") val itemName: String,
    @ColumnInfo(name = "isPriority") val isPriority: Boolean,
    @ColumnInfo(name = "isNeeded") val isNeeded: Boolean,
    @ColumnInfo(name = "isStaple") val isStaple: Boolean,
    /**
     * 1 iff `isx.lastPurchasedAt` falls inside the current session window,
     * computed in SQL so the repo doesn't need to re-thread `sessionStartMs`
     * to partition staples. Used by the picker repo: a staple that hasn't
     * been bought this session still counts as "on the list" for chip /
     * banner purposes; one bought this session moves to picked-up.
     */
    @ColumnInfo(name = "purchasedThisSession") val purchasedThisSession: Boolean,
)

File: ios/Storehop/Data/DB/StorePickerItemRow.swift
import Foundation
import GRDB

/// One row per (item, store) pair currently relevant to a Store Picker badge:
/// either still needed at this store, OR purchased there within the active
/// shopping session. Lets the picker render an "All set" affirmation on a
/// store after the user has checked off everything needed there.
///
/// The picker repository groups by `storeId` and produces one badge per
/// store. Multiple rows for the same `itemId` if it's tagged to multiple
/// stores.
struct StorePickerItemRow: FetchableRecord, Decodable, Hashable, Sendable {
    let storeId: String
    let itemId: String
    let itemName: String
    let isPriority: Bool
    let isNeeded: Bool
    let isStaple: Bool
    /// `true` iff `isx.lastPurchasedAt` falls inside the current session
    /// window — computed in SQL via a `CASE WHEN ... THEN 1 ELSE 0 END` so
    /// the repo doesn't need to re-thread `sessionStartMs` to partition
    /// staples. A priority staple that hasn't been bought this session
    /// still counts as "on the list" for chip / banner purposes; one bought
    /// this session moves to picked-up.
    let purchasedThisSession: Bool
}
